src/library/scala/collection/immutable/Queue.scala

```diff
@@ -100,6 +100,15 @@ class Queue[+A] protected(protected val in: List[A], protected val out: List[A])
     case _                               => super.:+(elem)(bf)
   }
 
+  override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
+    if (bf eq Queue.ReusableCBF) {
```

Comment:
@edmundnoble - Ugh, I'm sorry I overlooked this after the first comment.  As I guess we're all aware now, it's the `that: That` assumption that is wrong.  `That` is the type being produced, while `that` is an instance that is not parameterized by type.  It's a very unfortunate naming convention.